common/render.go

```diff
@@ -118,6 +118,42 @@ func (rs *RenderSystem) New(w *ecs.World) {
 }
 
 func (rs *RenderSystem) Add(basic *ecs.BasicEntity, render *RenderComponent, space *SpaceComponent) {
+	// Setting default shader
+	if render.shader == nil {
+		switch render.Drawable.(type) {
+		case Triangle:
+			render.shader = LegacyShader
+		case Circle:
+			render.shader = LegacyShader
+		case Rectangle:
+			render.shader = LegacyShader
+		case ComplexTriangles:
+			render.shader = LegacyShader
+		case Text:
+			render.shader = TextShader
+		default:
+			render.shader = DefaultShader
+		}
+	}
+
+	// This is to prevent users from using the wrong one
+	if render.shader == HUDShader {
+		switch render.Drawable.(type) {
+		case Triangle:
+			render.shader = LegacyHUDShader
+		case Circle:
+			render.shader = LegacyHUDShader
+		case Rectangle:
+			render.shader = LegacyHUDShader
+		case ComplexTriangles:
+			render.shader = LegacyHUDShader
+		case Text:
+			render.shader = TextHUDShader
+		default:
+			render.shader = HUDShader
+		}
+	}
+
```

Comment:
These set default shaders for those that have no shaders set, depending on the `Drawable`-type, so you don't explicitly have to say which shader you want to use, when you're using shapes / text